### PR TITLE
standard encoding for keys

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: apps/smp-server/Main.hs [hunk 1]
```diff
@@ -1,13 +1,15 @@
+{-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE OverloadedStrings #-}
 
 module Main where
 
 import Control.Monad (when)
-import Data.Attoparsec.ByteString.Char8 (Parser)
```
Comment (efim-poberezkin) [inline apps/smp-server/Main.hs:6]: Could we gradually switch to importing specific identifiers / module as qualified, please? (in this case the former but I'm just saying as a general suggestion)
It's just so much nicer to support afterwards w/t having to rely on external tools to figure out what comes where from.

File: package.yaml [hunk 2]
```diff
@@ -12,6 +12,9 @@ extra-source-files:
   - README.md
 
 dependencies:
+  - ansi-terminal == 0.10.*
```
Comment (efim-poberezkin) [inline package.yaml:15]: Is this added by a mistake?

**Review by efim-poberezkin — APPROVED** [review]

File: apps/smp-server/Main.hs [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+import Crypto.Store.PKCS8
+import Crypto.Store.X509
```
Comment (efim-poberezkin) [inline apps/smp-server/Main.hs:8]: Could we gradually switch to importing specific identifiers / module as qualified, please? (in this case the former but I'm just saying as a general suggestion)
It's just so much nicer to support afterwards w/t having to rely on external tools to figure out what comes where from.

File: src/Simplex/Messaging/Crypto.hs [hunk 4]
```diff
@@ -139,29 +165,23 @@ aesKeySize = 256 `div` 8
...
+          d = R.private_d pk
+      if d * d < n
+        then loop
+        else pure (PublicKey k, mkPrivateKey pk)
+
+rsaPrivateSize :: PrivateKey k => k -> Int
+rsaPrivateSize = R.public_size . R.private_pub . rsaPrivateKey
+
+validKeyPair :: FullKeyPair -> Bool
+validKeyPair (PublicKey k, FullPrivateKey pk) = k == R.private_pub pk
```
Comment (efim-poberezkin) [inline src/Simplex/Messaging/Crypto.hs:184]: Maybe just remove this and only store private key? It's FullPrivateKey stored now anyway. Or do we need pubkey saved separately for re-retrieval if it was lost by the server owner? I believe we discussed it but I don't remember the reason.

**Comment by efim-poberezkin** [comment]
"Encode keys" - lowercase title for consistency? - this now gets into changelog list

File: package.yaml (hunk 2 above)
Comment (epoberezkin) [reply package.yaml:15]: Colours in markdown

File: apps/smp-server/Main.hs (hunk 3 above)
Comment (epoberezkin) [reply apps/smp-server/Main.hs]: ok in this case, not as a principle though.

A common case when it is very inconvenient if you are using a lot of the imports and it either requires very long imports or creates a lot of visual noise.

**Pull request merged**